packages/dom-expressions/src/client.js

```diff
@@ -366,15 +364,16 @@ function assignProp(node, prop, value, prev, isSVG, skipRef, props) {
     (isChildProp = ChildProperties.has(prop)) ||
     (!isSVG &&
       ((propAlias = getPropAlias(prop, node.tagName)) || (isProp = Properties.has(prop)))) ||
-    (isCE = (node.nodeName.includes("-") || 'is' in props))
+    (isCE = node.nodeName.includes("-") || "is" in props)
   ) {
     if (forceProp) {
       prop = prop.slice(5);
       isProp = true;
     } else if (isHydrating(node)) return value;
     if (prop === "class" || prop === "className") className(node, value);
-    else if (isCE && !isProp && !isChildProp) node[toPropertyName(prop)] = value;
-    else node[propAlias || prop] = value;
+    else if (isCE && !isProp && !isChildProp) {
+      node.setAttribute(prop, value);
```

Comment:
This is nice, unlike React's which forces a string coercion of `value` before passing it to `setAttribute`. The native DOM `setAttribute` method already will do that coercion, but! a custom element class can override the `setAttribute` method to also handle non-string values and for example map them directly to JS properties, which is not possible with React at time of this writing.

This means that, even if Solid JSX calls `el.setAttribute`, a custom element can itself be designed to pass the value directly to JS and skip the native `setAttribute`, therefore achieving the same thing as `prop:` without the user having to use `prop:` syntax. The A-Frame library, for example, overrides `setAttribute` like this for all its elements. I'm contemplating adding this to Lume Element. In my case, I could map JS value types to the JSX attribute types. But other people's CE libraries don't have to, they can map only string types for non-`prop:` attributes for example.

Reply:
In fact, had React at least removed that stupid string coercion, they'd have largely eliminated the need for React 19 CE support because all CEd would have had a way to accept non-string values.

But they didn't listen.

Reply:
Overriding `setAttribute` is interesting and something I hadn't thought of. I'm sure it has other implications/inconsistency but it definitely is interesting.